kyuubi-server/src/main/scala/org/apache/kyuubi/server/metadata/MetadataManager.scala

```diff
@@ -37,44 +37,55 @@ class MetadataManager extends AbstractService("MetadataManager") {
   private var _metadataStore: MetadataStore = _
 
   // Visible for testing.
-  private[metadata] val identifierRequestsRetryRefs =
+  private[metadata] val identifierRequestsAsyncRetryRefs =
     new ConcurrentHashMap[String, MetadataRequestsRetryRef]()
 
   // Visible for testing.
-  private[metadata] val identifierRequestsRetryingCounts =
+  private[metadata] val identifierRequestsAsyncRetryingCounts =
     new ConcurrentHashMap[String, AtomicInteger]()
 
-  private val requestsRetryTrigger =
-    ThreadUtils.newDaemonSingleThreadScheduledExecutor("metadata-requests-retry-trigger")
+  private lazy val requestsRetryInterval =
+    conf.get(KyuubiConf.METADATA_REQUEST_RETRY_INTERVAL)
 
-  private var requestsRetryExecutor: ThreadPoolExecutor = _
+  private lazy val requestsAsyncRetryEnabled =
+    conf.get(KyuubiConf.METADATA_REQUEST_ASYNC_RETRY_ENABLED)
```

Comment:
so this is the key change that introduce a config to control if we should retry ?

Reply:
yea